plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -54,7 +54,7 @@ export const EntityRefLink = forwardRef<any, EntityRefLinkProps>(
     let name;
 
     if (typeof entityRef === 'string') {
-      const parsed = parseEntityRef(entityRef);
+      const parsed = parseEntityRef(entityRef, { defaultKind: defaultKind });
```

Comment:
We're feeling this behavior is a bit too risky compared to what one might expect, and we'd rather be a bit strict in how the input is treated. You can always parse the entity ref outside of the component using a default kind and then pass in the compound ref instead.

The `defaultKind` is really just for the display of the link, rather than the ref that's passed in. Perhaps that's something we could highlight better in docs though, would you like to change this PR to add that instead?

Should add that this is our current opinion either way, if you have a strong argument for why it would be better to have the `defaultKind` affect the ref parsing we're happy to discuss it further :grin:

Reply:
> compared to what one might expect

Yeh, my expectation was based on how the `defaultKind` parameter works with other APIs that deal with entity refs.

> The defaultKind is really just for the display of the link, rather than the ref that's passed in.

What's the difference in practice though? Ultimately this is a signal from the called that this is the kind to use unless otherwise specified. I don't see the practical benefit of not using it here, and it doesn't seem like a risk change either.

Reply:
It's a signal that if the default kind matches the kind of the ref, then there's no need to display the kind. It's more about signaling the context in which the ref is being displayed, rather than what it is.